presentation-compiler/src/main/dotty/tools/pc/WithCompilationUnit.scala

```diff
@@ -0,0 +1,104 @@
+package dotty.tools.pc
+
+import scala.language.unsafeNulls
+
+import java.nio.file.Paths
+
+import scala.meta as m
+
+import scala.meta.internal.metals.CompilerOffsetParams
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.VirtualFileParams
+
+import dotty.tools.dotc.core.Contexts.*
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.core.NameOps.*
+import dotty.tools.dotc.core.Symbols.*
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.pc.utils.InteractiveEnrichments.*
+
+class WithCompilationUnit(
+    val driver: InteractiveDriver,
+    params: VirtualFileParams,
+):
+  val uri = params.uri()
+  val filePath = Paths.get(uri)
+  val sourceText = params.text
+  val text = sourceText.toCharArray()
+  val source =
+    SourceFile.virtual(filePath.toString, sourceText)
+  driver.run(uri, source)
+  given ctx: Context = driver.currentCtx
+
+  val unit = driver.currentCtx.run.units.head
```

Comment:
quickfix:
```suggestion
  private val run = driver.currentCtx.run
  val unit = run.units.head
```
(I was incorrect before about the java compat thing). This is ending up a bit hard to minimize since it both depends on the bootstrapped scala compiler and has to be compiled by it (nonbootstrapped works fine). My attempts to have it crash with `scala3-bootstrapped/scalac -with-compiler` are also proving unsuccessful. I might not be able to come up with the minimization after all, but the quick fix should help